logstash-core/lib/logstash/api/rack_app.rb

```diff
@@ -8,9 +8,84 @@
 module LogStash
   module Api
     module RackApp
-      def self.app
+      # Cabin is not compatible with CommonLogger, and this gives us more control anyway
+      class ApiLogger
+        LOG_MESSAGE = "API HTTP Request".freeze
+        
+        def initialize(app, logger)
+          @app = app
+          @logger = logger
+        end
+
+        def call(env)
+          res = @app.call(env)
+          status, headers, body = res
+          
+          if status >= 500 && status < 600
+            @logger.warn? && @logger.warn(LOG_MESSAGE, log_metadata(status, env))                      
+          else          
+            @logger.info? && @logger.info(LOG_MESSAGE, log_metadata(status, env))                      
+          end
+
+          res          
+        end
+
+        def log_metadata(status, env)
+          {
+            :status => status,
+            :method => env["REQUEST_METHOD"],
```

Comment:
can we automate this, might be using an inject loop? I ask because it makes sense to actually log all the event.